client/SDL3/sdl_clip.cpp

```diff
@@ -0,0 +1,765 @@
+	ClipboardLockGuard(ClipboardLockGuard&& other) = delete;
+
+	ClipboardLockGuard& operator=(const ClipboardLockGuard& rhs) = delete;
+	ClipboardLockGuard& operator=(ClipboardLockGuard&& rhs) = delete;
+
+	~ClipboardLockGuard()
+	{
+		ClipboardUnlock(_clipboard);
+	}
+
+  private:
+	wClipboard* _clipboard;
+};
+
+static bool operator<(const CLIPRDR_FORMAT& lhs, const CLIPRDR_FORMAT& rhs)
+{
+	return (lhs.formatId < rhs.formatId);
+}
+static bool operator>(const CLIPRDR_FORMAT& lhs, const CLIPRDR_FORMAT& rhs)
+{
+	return (lhs.formatId > rhs.formatId);
+}
+static bool operator==(const CLIPRDR_FORMAT& lhs, const CLIPRDR_FORMAT& rhs)
+{
+	return (lhs.formatId == rhs.formatId);
+}
+
+sdlClip::sdlClip(SdlContext* sdl)
+    : _sdl(sdl), _file(cliprdr_file_context_new(this)), _log(WLog_Get(TAG)),
+      _system(ClipboardCreate()), _event(CreateEventA(nullptr, TRUE, FALSE, nullptr))
+{
+	WINPR_ASSERT(sdl);
+}
+
+sdlClip::~sdlClip()
+{
+	cliprdr_file_context_free(_file);
+	ClipboardDestroy(_system);
+	CloseHandle(_event);
+}
+
+BOOL sdlClip::init(CliprdrClientContext* clip)
+{
+	WINPR_ASSERT(clip);
+	_ctx = clip;
+	clip->custom = this;
+	_ctx->MonitorReady = sdlClip::MonitorReady;
+	_ctx->ServerCapabilities = sdlClip::RecieveServerCapabilities;
+	_ctx->ServerFormatList = sdlClip::ReceiveServerFormatList;
+	_ctx->ServerFormatListResponse = sdlClip::ReceiveFormatListResponse;
+	_ctx->ServerFormatDataRequest = sdlClip::ReceiveFormatDataRequest;
+	_ctx->ServerFormatDataResponse = sdlClip::ReceiveFormatDataResponse;
+
+	return cliprdr_file_context_init(_file, _ctx);
+}
+
+BOOL sdlClip::uninit(CliprdrClientContext* clip)
+{
+	WINPR_ASSERT(clip);
+	if (!cliprdr_file_context_uninit(_file, _ctx))
+		return FALSE;
+	_ctx = nullptr;
+	clip->custom = nullptr;
+	return TRUE;
+}
+
+bool sdlClip::handle_update()
+{
+	if (!_ctx || !_sync)
+		return true;
+
+	clearServerFormats();
+
+	std::string mime_uri_list = "text/uri-list";
+	std::string mime_html = "text/html";
+
+	std::vector<std::string> mime_bitmap = { "image/bmp", "image/x-bmp", "image/x-MS-bmp",
+		                                     "image/x-win-bitmap" };
+	std::string mime_webp = "image/webp";
+	std::string mime_png = "image/png";
+	std::string mime_jpeg = "image/jpeg";
+	std::string mime_tiff = "image/tiff";
+	std::vector<std::string> mime_images = { mime_webp, mime_png, mime_jpeg, mime_tiff };
+
+	std::vector<std::string> clientFormatNames;
+	std::vector<CLIPRDR_FORMAT> clientFormats;
+	if (SDL_HasClipboardText())
+	{
+		clientFormats.push_back({ CF_TEXT, nullptr });
+		clientFormats.push_back({ CF_OEMTEXT, nullptr });
+		clientFormats.push_back({ CF_UNICODETEXT, nullptr });
+	}
+	if (SDL_HasClipboardData(mime_html.c_str()))
+		clientFormatNames.push_back(type_HtmlFormat);
+
+	for (auto& mime : mime_bitmap)
+	{
+		if (SDL_HasClipboardData(mime.c_str()))
+		{
+			clientFormats.push_back({ CF_DIB, nullptr });
+			clientFormats.push_back({ CF_DIBV5, nullptr });
```

Comment:
If I remember correctly, the conversion from and to `CF_DIBV5` isn't implemented in `wClipboard`, so this will unlikely work (`CF_DIB` works though).

Reply:
it is handled like that for other clients as well.

looks like the format is unused by windows nowadays?

Reply:
I´ll keep it this way and (if need arises) we´ll implement the format